dotCMS/src/main/java/com/dotmarketing/fixtask/tasks/FixTask00095DeleteOrphanRelationships.java

```diff
@@ -0,0 +1,173 @@
+                    Audit.setTableName("relationship");
+                    Audit.setDatetime(new Date());
+                    Audit.setRecordsAltered(total);
+                    Audit.setAction("task 95: DeleteOrphanRelationships");
+                    HibernateUtil.save(Audit);
+                    HibernateUtil.commitTransaction();
+                    returnValue.add(FixAssetsProcessStatus.getFixAssetsMap());
+                    Logger.debug(
+                            FixTask00095DeleteOrphanRelationships.class,
+                            "Ending DeleteOrphanRelationships");
+                }
+            } catch (Exception e) {
+                Logger.debug(
+                        FixTask00095DeleteOrphanRelationships.class,
+                        "There was a problem during DeleteOrphanRelationships",
+                        e);
+                HibernateUtil.rollbackTransaction();
+                FixAssetsProcessStatus.setActual(-1);
+            } finally {
+                FixAssetsProcessStatus.stopProgress();
+            }
+        }
+        return returnValue;
+    }
+
+    @Override
+    public List<Map<String, String>> getModifiedData() {
+        if (modifiedData.size() > 0) {
+            XStream _xstream = new XStream(new DomDriver());
+            Date date = new Date();
+            SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy_HH-mm-ss");
+            String lastmoddate = sdf.format(date);
+            File _writing = null;
+            if (!new File(ConfigUtils.getBackupPath() + File.separator
+                    + "fixes").exists()) {
+                new File(ConfigUtils.getBackupPath() + File.separator + "fixes")
+                        .mkdirs();
+            }
+            _writing = new File(ConfigUtils.getBackupPath() + File.separator
+                    + "fixes" + java.io.File.separator + lastmoddate + "_"
+                    + "FixTask00095DeleteOrphanRelationships" + ".xml");
+
+            BufferedOutputStream _bout = null;
+            try {
+                _bout = new BufferedOutputStream(new FileOutputStream(_writing));
+            } catch (FileNotFoundException e) {
+                Logger.error(this, "Could not write to Fix Task status file.");
+            }
+            _xstream.toXML(modifiedData, _bout);
+        }
+        return modifiedData;
+    }
+
+    @Override
+    public boolean shouldRun() {
+        Logger.debug(CMSMaintenanceFactory.class, "Running query for relationships: "
+                + VERIFICATION_QUERY);
+        DotConnect dc = new DotConnect();
+        dc.setSQL(VERIFICATION_QUERY);
+        List<Map<String, Object>> inodesInStructure = null;
+        try {
+            inodesInStructure = dc.loadObjectResults();
+        } catch (DotDataException e) {
+            Logger.error(this, e.getMessage(), e);
+        }
+        Logger.debug(CMSMaintenanceFactory.class,
+                "Found " + inodesInStructure.size() + " invalid inodes.");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "inodesInStructure" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)